### PR TITLE
JobIntegrationTest: Increase the restConfig QPS and Burst parameters

### PR DESCRIPTION
**Pull request by tenzen-y**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
As we discussed in the following PR, we decided to increase restConfig QPS and Burst to avoid flaky testing due to client rate-limiting errors.

https://github.com/kubernetes/kubernetes/pull/127078

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by mimowo — COMMENTED** [review]
LGTM, I would like to also bump it for `BenchmarkLargeIndexedJob` and `BenchmarkLargeFailureHandling` (to align the values). 

I added it for `BenchmarkLargeFailureHandling`, but this was just copy-paste from `BenchmarkLargeIndexedJob`. WDYT @alculquicondor ?

**Review by mimowo — COMMENTED** [review]
/lgtm

**Comment by tenzen-y** [comment]
> LGTM, I would like to also bump it for `BenchmarkLargeIndexedJob` and `BenchmarkLargeFailureHandling` (to align the values).
> 
> I added it for `BenchmarkLargeFailureHandling`, but this was just copy-paste from `BenchmarkLargeIndexedJob`. WDYT @alculquicondor ?

That is a good point. I updated QPS and Burst for both testing now. If @alculquicondor has any concerns, I can revert it.

File: test/integration/job/job_test.go [hunk 1]
```diff
@@ -3304,8 +3307,8 @@ func TestElasticIndexedJob(t *testing.T) {
 // also faster to track, as they need less API calls.
 func BenchmarkLargeIndexedJob(b *testing.B) {
 	closeFn, restConfig, clientSet, ns := setup(b, "indexed")
-	restConfig.QPS = 100
-	restConfig.Burst = 100
+	restConfig.QPS = restConfigQPS
```
Comment (alculquicondor) [inline test/integration/job/job_test.go:3310]: such a high QPS makes sense for integration, as we are trying to measure correctness.

But in a benchmark, we want to make sure that the Job controller is fast enough when the QPS is limited.

Comment (tenzen-y) [reply test/integration/job/job_test.go]: That makes sense.
I reverted those, and added comments the reason.
PTAL.

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

File: test/integration/job/job_test.go (hunk 1 above)
Comment (mimowo) [reply test/integration/job/job_test.go]: Makes sense, thanks for the suggestion and the update.

**Comment by mimowo** [comment]
LGTM

**Pull request merged**